src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,8 +1,30 @@
-export const TodoFilter = () => (
+import React from 'react';
+
+type Props = {
+  onSetQuery(newQuery: string): void;
+  onSetFilter(newFilter: string): void;
+  query: string;
+  filter: string;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  onSetQuery,
+  onSetFilter,
+  query,
+  filter,
+}) => (
   <form className="field has-addons">
     <p className="control">
       <span className="select">
-        <select data-cy="statusSelect">
+        <select
+          value={filter}
+          data-cy="statusSelect"
+          onChange={
+            (event: React.ChangeEvent<HTMLSelectElement>) => onSetFilter(
+              event.target.value,
+            )
+          }
```

Comment:
I think that setting the event type is not necessary if you are working with it directly in the event handler